src/interface.jl

```diff
@@ -31,111 +46,201 @@ ADBackend(::Val) = error("The requested AD backend is not available. Make sure t
 ######################
 # Bijector interface #
 ######################
-"Abstract type for a bijector."
-abstract type AbstractBijector end
+"""
+
+Abstract type for a transformation.
+
+# Implementing
+
+A subtype of `Transform` of should at least implement [`transform(b, x)`](@ref).
+
+If the `Transform` is also invertible:
+- Required:
+  - [`invertible`](@ref): should return instance of [`Invertible`](@ref) or [`NotInvertible`](@ref).
+  - _Either_ of the following:
+    - `transform(::Inverse{<:MyTransform}, x)`: the `transform` for its inverse.
+    - `InverseFunctions.inverse(b::MyTransform)`: returns an existing `Transform`.
+  - [`logabsdetjac`](@ref): computes the log-abs-det jacobian factor.
+- Optional:
+  - `with_logabsdet_jacobian`: `transform` and `logabsdetjac` combined. Useful in cases where we
+    can exploit shared computation in the two.
+
+For the above methods, there are mutating versions which can _optionally_ be implemented:
+- [`with_logabsdet_jacobian!`](@ref)
+- [`logabsdetjac!`](@ref)
+- [`with_logabsdet_jacobian!`](@ref)
+"""
+abstract type Transform end
+
+(t::Transform)(x) = transform(t, x)
+
+Broadcast.broadcastable(b::Transform) = Ref(b)
+
+"""
+    transform(b, x)
+
+Transform `x` using `b`, treating `x` as a single input.
+"""
+transform(f::Function, x) = f(x)
+transform(t::Transform, x) = first(with_logabsdet_jacobian(t, x))
+
+"""
+    transform!(b, x[, y])
 
-"Abstract type of bijectors with fixed dimensionality."
-abstract type Bijector{N} <: AbstractBijector end
+Transform `x` using `b`, storing the result in `y`.
 
-dimension(b::Bijector{N}) where {N} = N
-dimension(b::Type{<:Bijector{N}}) where {N} = N
+If `y` is not provided, `x` is used as the output.
+"""
+transform!(b, x) = transform!(b, x, x)
+transform!(b, x, y) = (y .= transform(b, x))
```

Comment:
Is broadcasting the best fallback here? Seems at least not obvious from the function name that the arguments are broadcasted. Maybe rather
```suggestion
transform!(b, x, y) = copyto!(y, transform(b, x))
```
?